lib/queues.js

```diff
@@ -341,11 +338,11 @@ class Queue {
           // Maybe we were requested to shut down while waiting
           // for jobs. Don't process, try to release them instead.
           if (this._processing) {
-            this._activeJobs += jobs.length;
+            this._server.incrementActiveJobs(jobs.length);
             jobs.forEach(job => {
               this._runAndDestroy(client, job)
                 .catch(() => { })
-                .then(() => --this._activeJobs);
+                .then(() => this._server.incrementActiveJobs(-1));
```

Comment:
If you increment X by Y, the result must be higher than X. Otherwise, X was not incremented.